packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.js

```diff
@@ -13,16 +13,27 @@ import ssrPrepass from 'react-ssr-prepass'
 const isServerSide = typeof window === 'undefined'
 const STATE_KEY = '__reactQuery'
 
-export const withReactQuery = (Wrapped) => {
+export const withReactQuery = (Wrapped, options = {}) => {
     /* istanbul ignore next */
     const wrappedComponentName = Wrapped.displayName || Wrapped.name
+    const queryClientConfig = options.queryClientConfig || {
+        defaultOptions: {
+            queries: {
+                retry: !isServerSide
+            },
+            mutations: {
+                retry: !isServerSide
+            }
+        }
+    }
```

Comment:
The constructor `new QueryClient(queryClientConfig))` expects `queryClientConfig` type to be `QueryClientConfig`.

i.e.
```suggestion
    const queryClientConfig = options.defaultQueryClientConfig || {
        defaultQueryClientConfig: {
            defaultOptions: {
                queries: {
                    retry: !isServerSide
                },
                mutations: {
                    retry: !isServerSide
                }
            }
        }
    }
```

If we expect users to use it in their `_app-config/index.tsx` like this:

`packages/test-commerce-sdk-react/app/components/_app-config/index.tsx`
```
import {DefaultOptions, QueryClientConfig} from '@tanstack/react-query'

const defaultOptions: DefaultOptions = {
    queries: {
        staleTime: 1000 * 20 // default 0
    }
}

const defaultQueryClientConfig: QueryClientConfig = {defaultOptions}

...

export default withReactQuery(AppConfig, {defaultQueryClientConfig})
```

Reply:
I'm not sure I'm following your comment. As per the docs a client is defined like below:

```
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: Infinity,
    },
  },
})
```

Reply:
Ignore my comment on the format. I updated the comment to be just a code example of how users could override the default global configuration in their user space with this approach.

Note that the default global configuration defined by the user will be used for server-side and client-side queries.
Thus with this approach users can't choose to adjust the number of retries client-side without affecting the number of retries server-side.

Reply:
Also, I just found out that if we set `retry: true` 

> will **infinitely** retry failing requests.

I don't think we want to do that for all the client-side requests.

https://tanstack.com/query/v4/docs/guides/query-retries#:~:text=Setting%20retry%20%3D%20true%20will%20infinitely%20retry%20failing%20requests.

Reply:
nice catch with the infinite thing

Reply:
Ah yes, good catch @adamraya about the boolean value for `retry`. I faced that gotcha too in my PR #718 . 

So what we'll need in this case is to use a _function_ instead for the `retry` option: `retry = (failureCount, error) => ...`. In the function:
- we'd check whether on server side or not
- on the client side, also use the `failureCount` to stop retrying after certain number of retries.

Reply:
The configuration is in the user-space now. So there will be no tricky issues with setting the values to true using the logic I had in a previous iteration of this pr (e.g. retry: !!isServerSide)